packages/caver-wallet/src/keyringContainer.js

```diff
@@ -0,0 +1,308 @@
+    GNU Lesser General Public License for more details.
+
+    You should have received a copy of the GNU Lesser General Public License
+    along with the caver-js. If not, see <http://www.gnu.org/licenses/>.
+*/
+
+const _ = require('lodash')
+const Keyring = require('./keyring/keyring')
+const TransactionHasher = require('../../caver-transaction/src/transactionHasher/transactionHasher')
+const { KEY_ROLE } = require('./keyring/keyringHelper')
+const utils = require('../../caver-utils/src')
+
+/**
+ * representing a Keyring container which manages keyrings.
+ * @class
+ */
+class KeyringContainer {
+    /**
+     * creates a keyringContainer.
+     * @param {Array.<Keyring>} keyrings - The keyrings to be managed in KeyringContainer.
+     */
+    constructor(keyrings) {
+        keyrings = keyrings || []
+        this._length = 0
+        this._addressKeyringMap = new Map()
+
+        // add keyrings to keyringContainer
+        for (const keyring of keyrings) {
+            this.add(keyring)
+        }
+    }
+
+    /**
+     * @type {number}
+     */
+    get length() {
+        return this._length
+    }
+
+    /**
+     * generates keyrings in the keyringContainer with randomly generated key pairs.
+     *
+     * @param {number} numberOfKeyrings The number of accounts to create.
+     * @param {string} [entropy] A random string to increase entropy. If undefined, a random string will be generated using randomHex.
+     * @return {KeyringContainer}
+     */
+    generate(numberOfKeyrings, entropy) {
+        for (let i = 0; i < numberOfKeyrings; ++i) {
+            this.add(Keyring.generate(entropy))
+        }
+        return this
+    }
+
+    /**
+     * creates a keyring instance with given parameters and adds it to the keyringContainer.
+     *
+     * @param {string} address The address of the keyring.
+     * @param {string|Array.<string>|Array.<Array.<string>>} key Private key string(s) to use in keyring. If different keys are used for each role, key must be defined as a two-dimensional array.
+     * @return {Keyring}
+     */
+    newKeyring(address, key) {
+        // The format of key parameter can be
+        // 1. single private key string   => `0x{private key}`
+        // 2. multiple private key string =>[`0x{private key}`, `0x{private key}`, ...]
+        // 3. role based private keys     => [[`0x{private key}`, `0x{private key}`, ...], [], [`0x{private key}`]]
+
+        let keyring
+
+        if (_.isString(key)) keyring = Keyring.createWithSingleKey(address, key)
+
+        if (_.isArray(key)) {
+            if (key.length === 0) throw new Error(`Insufficient private key information: Empty array`)
+            if (_.isArray(key[0])) {
+                keyring = Keyring.createWithRoleBasedKey(address, key)
+            } else {
+                keyring = Keyring.createWithMultipleKey(address, key)
+            }
+        }
+
+        if (!(keyring instanceof Keyring)) throw new Error(`Unsupported type : ${typeof key}`)
```

Comment:
What about printing the value of `key` as well? It is much better for debugging if we know the value.

Reply:
I changed error message!